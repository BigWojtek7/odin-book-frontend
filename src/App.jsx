import { Outlet } from 'react-router-dom';
import Navbar from './components/Navbar/Navbar';
import { useEffect, useState } from 'react';
import getRequestWithNativeFetch from './utils/fetchApiGet';

function App() {
  const [user, setUser] = useState({});

  const currentToken = localStorage.getItem('token');
  const [token, setToken] = useState(currentToken);
  const [isLoading, setIsLoading] = useState(false);
  useEffect(() => {
    if (token) {
      const fetchDataForUsers = async () => {
        try {
          const url = `${import.meta.env.VITE_BACKEND_URL}/users/user`;
          const headers = {
            Authorization: token,
          };
          const userData = await getRequestWithNativeFetch(url, headers);
          setUser(userData);
        } catch (err) {
          console.log(err);
        }
      };
      fetchDataForUsers();
    }
    return () => {
      setUser([]);
    };
  }, [token]);
  return (
    <>
      <Navbar token={token} setToken={setToken} />
      <main>
        <Outlet context={[token, setToken, user, isLoading, setIsLoading]} />
      </main>
    </>
  );
}

export default App;
