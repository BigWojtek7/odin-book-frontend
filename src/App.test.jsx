// import { describe, it, expect } from 'vitest';

describe('something truthy and falsy', () => {
  it('true to be true', () => {
    expect(true).toBe(true);
  });

  it('false to be false', () => {
    expect(false).toBe(false);
  });
});

import { render, screen } from '@testing-library/react';

import App from './App';
import { BrowserRouter } from 'react-router-dom';

describe('App', () => {
  it('renders whole app', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  it('renders paragraph', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    expect(screen.getByText('FriendsBook')).toBeInTheDocument();
  });
});
