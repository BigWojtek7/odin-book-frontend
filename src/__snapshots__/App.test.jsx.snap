// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders whole app 1`] = `
<div>
  <div
    class="_navHeader_278c00"
  >
    <button
      class="_btnToggle_278c00"
    >
      <svg
        role="presentation"
        style="width: 2.0999999999999996rem; height: 2.0999999999999996rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M3,6H21V8H3V6M3,11H21V13H3V11M3,16H21V18H3V16Z"
          style="fill: currentColor;"
        />
      </svg>
    </button>
    <p
      class="_logo_278c00"
    >
      FriendsBook
    </p>
    <nav
      class="_nav_278c00 "
    >
      <ul
        class="_navList_278c00"
      >
        <li>
          <a
            class="_navLink_278c00"
            href="/sign-up"
          >
            <svg
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15,14C12.33,14 7,15.33 7,18V20H23V18C23,15.33 17.67,14 15,14M6,10V7H4V10H1V12H4V15H6V12H9V10M15,12A4,4 0 0,0 19,8A4,4 0 0,0 15,4A4,4 0 0,0 11,8A4,4 0 0,0 15,12Z"
                style="fill: currentColor;"
              />
            </svg>
            SignUp
          </a>
        </li>
        <li>
          <a
            class="_navLink_278c00"
            href="/login"
          >
            <svg
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 7L9.6 8.4L12.2 11H2V13H12.2L9.6 15.6L11 17L16 12L11 7M20 19H12V21H20C21.1 21 22 20.1 22 19V5C22 3.9 21.1 3 20 3H12V5H20V19Z"
                style="fill: currentColor;"
              />
            </svg>
            Login
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <main />
</div>
`;
